activesupport/lib/active_support/cache/coder.rb

```diff
@@ -88,6 +90,28 @@ def load(dumped)
 
         MARSHAL_SIGNATURE = "\x04\x08".b.freeze
 
+        class LazyEntry < Cache::Entry
+          def initialize(coder, payload, type, **options)
+            super(payload, **options)
+            @coder = coder
+            @type = type
+          end
+
+          def value
+            if @type
+              @value = @coder._load_payload(@value, @type)
```

Comment:
Not a big fan of that `_load_payload`.

Could we pass both the `serializer` and `compressor`? If the payload isn't compressed, we pass `compressor = nil`.

How does that sound?

Reply:
If that is what you prefer, we can do that.

The reason I organized it as `_load_payload` is the same reason I originally used a thunk: this code is essentially part of `load`, it is merely delayed.  Encapsulating the code in `_load_payload` allows it to be near to `load` so they can be understood together.

I've pushed a commit that passes `serializer` and `compressor`, and `compressor = nil` when the payload isn't compressed.  In my opinion, the code is disjointed and less clear.

If you can tell me what you dislike about `_load_payload`, perhaps I can iterate on it.  If you dislike the name, I can change it.  If you dislike `LazyEntry` delegating to `@coder._load_payload`, I can pass (a cached copy of) `method(:_load_payload)` instead.

Reply:
> If you can tell me what you dislike about `_load_payload`,

A few things:

  - It increase the interface, particularly with something that is public but private `_load`.
  - I like that coder, serializers etc have a simple and standard `load/dump`, `inflate/deflate` API.
  - It makes the entry and coder reflation two way, I prefer when reference can go only in one direction.

Looking at your last commit now.

Reply:
> * It increase the interface, particularly with something that is public but private `_load`.
>
> * I like that coder, serializers etc have a simple and standard `load/dump`, `inflate/deflate` API.

These could be solved by passing something like `method(:_load_payload)` instead of the coder instance.

> * It makes the entry and coder reflation two way, I prefer when reference can go only in one direction.

In terms of `LazyEntry` knowing about `Cache::Coder`, passing `method(:_load_payload)` would also abstract that relationship.

In terms of references and lifetimes, `LazyEntry` must still hold references to call `@serializer.load` and `@compressor.inflate`, so there isn't much difference either way.